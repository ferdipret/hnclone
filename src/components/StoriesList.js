import React from 'react'
import Story from './Story'
import '../styles/stories-list.css'

function StoriesList({ stories }) {
  return (
    stories && (
      <div className="list-container">
        {stories.map(story => <Story key={story.id} {...story} />)}
      </div>
    )
  )
}

export default StoriesList
