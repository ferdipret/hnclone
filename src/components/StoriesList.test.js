import React from 'react'
import { shallow } from 'enzyme'
import StoriesList from './StoriesList'

describe('StoriesList', () => {
  const storiesList = shallow(<StoriesList />)
  it('renders properly', () => {
    expect(storiesList).toMatchSnapshot()
  })
})
