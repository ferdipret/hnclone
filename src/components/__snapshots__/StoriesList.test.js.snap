// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesList renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StoriesList
    stories={
        Array [
            Object {
              "id": 1,
              "name": "Foo",
            },
            Object {
              "id": 2,
              "name": "Bar",
            },
            Object {
              "id": 3,
              "name": "Baz",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Story
          id={1}
          name="Foo"
/>,
        <Story
          id={2}
          name="Bar"
/>,
        <Story
          id={3}
          name="Baz"
/>,
      ],
      "className": "list-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "id": 1,
          "name": "Foo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "id": 2,
          "name": "Bar",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "id": 3,
          "name": "Baz",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Story
            id={1}
            name="Foo"
/>,
          <Story
            id={2}
            name="Bar"
/>,
          <Story
            id={3}
            name="Baz"
/>,
        ],
        "className": "list-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "id": 1,
            "name": "Foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "id": 2,
            "name": "Bar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "id": 3,
            "name": "Baz",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
